base_comment_template/models/base_comment_template.py

```diff
@@ -120,6 +120,8 @@ def name_get(self):
     def _search_model_ids(self, operator, value):
         # We cannot use model_ids.model in search() method to avoid recursion.
         allowed_items = (
-            self.sudo().search([]).filtered(lambda x: x.model_ids.model == value)
+            self.sudo()
+            .search([])
+            .filtered(lambda x: value in x.model_ids.mapped("model"))
```

Comment:
@victoralmau this was prepared for models split by commas?

@AaronHForgeFlow in any case, you should split the values by comma, and do exact match. If not, the model `purchase.order` will match with for example `purchase.order.line`

Reply:
model_ids is computed correctly, it has the 3 ir.model records. But then I get singleton error without this change.

Reply:
The mapped is a list, The filtered is checking the string with the model is in the list. The complete string must match an element in the list, otherwise it will be filtered out, I think:

```
>>> list = ['sale.order', 'purchase.order']
>>> 'purchase.order' in list
True
>>> 'purchase.order.line' in list
False

```

Reply:
Yes, the original idea is to be able to define several models separated by comma (`model_ids` will be self-defined).

Although the proposal is correct, I think this https://github.com/OCA/reporting-engine/blob/15.0/base_comment_template/models/comment_template.py#L35 would have to be adapted as well.

Reply:
Thank you @victoralmau, you are right. I adapted that part as well and it seems to be working when putting several models in comment templates and trying to get them in records for one of those models. They don't show if the record is for a model that is not in the template.